# Paul: This is a good start on your lab notebook, but ideally you would have better annotation for each entry. When you read this work a year or 2 from now, or if you were to share it with a collaborator, the notebook should include enough rationale and description of what you're doing and why so that it is interpretable. A bit more organization and annotation would increase its value to you for the future, and help with documentation purposes when it comes to writing up your work for publication, etc. --Steve


## Author: Paul Cronin     
## Ecological Genomics:   

### Overall Description of notebook      

This notebook will catalog my entries throughout the semester. This includes my anlyses, data and code.


### Date started: (2018-Jan-24)   
### Date end:   (year-month-day)    

### Philosophy   
Science should be reproducible and one of the best ways to achieve this is by logging research activities in a notebook. Because science/biology has increasingly become computational, it is easier to document computational projects in an electronic form, which can be shared online through Github.    

### Helpful features of the notebook     

**It is absolutely critical for your future self and others to follow your work.**     

* The notebook is set up with a series of internal links from the table of contents.    
* All notebooks should have a table of contents which has the "Page", date, and title (information that allows the reader to understand your work).     
* Also, one of the perks of keeping all activities in a single document is that you can **search and find elements quickly**.     
* Lastly, you can share specific entries because of the three "#" automatically creates a link when the notebook renders on github.      


<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.  


### Table of contents for 60 entries (Format is *Page: Date(with year-month-day). Title*)        
* [Page 1: 2018-01-24](#id-section1). Intro to Github, RMarkdown, and UNIX command-line
* [Page 2: 2018-01-29](#id-section2). Working with RNA-seq data
* [Page 3: 2018-01-31](#id-section3). RNA-seq cont.
* [Page 4: 2018-02-5](#id-section4). RNA-seq cont.
* [Page 5: 2018-02-7](#id-section5). RNA-seq cont.
* [Page 6: 2018-02-12](#id-section6). RNA-seq cont.
* [Page 7: 2018-02-14](#id-section7). RNA-seq cont. DESEq2
* [Page 8: 2018-02-21](#id-section8). RNA-seq cont. WGCNA
* [Page 9: 2018-02-26](#id-section9). Pop Genomics: Intro
* [Page 10: 2018-02-28](#id-section10). Pop Genomics: Diversity and sfs
* [Page 11: 2018-03-5](#id-section11). Pop Genomics: Missed Class
* [Page 12: 2018-03-7](#id-section12). Pop Genomics: Missed Class
* [Page 13: 2018-03-19](#id-section13). Pop Genomics: Admixture Analysis
* [Page 14: 2018-03-21](#id-section14). Pop Genomics: Fst Outlier Analysis
* [Page 15: 2018-03- 26](#id-section15). Pop Genomics: Fst Outlier Analysis cont.
* [Page 16:](#id-section16). Pop Genomics Homework 2
* [Page 17:](#id-section17). Group Project
* [Page 18:](#id-section18).
* [Page 19:](#id-section19).
* [Page 20:](#id-section20).
* [Page 21:](#id-section21).
* [Page 22:](#id-section22).
* [Page 23:](#id-section23).
* [Page 24:](#id-section24).
* [Page 25:](#id-section25).
* [Page 26:](#id-section26).
* [Page 27:](#id-section27).
* [Page 28:](#id-section28).
* [Page 29:](#id-section29).
* [Page 30:](#id-section30).
* [Page 31:](#id-section31).
* [Page 32:](#id-section32).
* [Page 33:](#id-section33).
* [Page 34:](#id-section34).
* [Page 35:](#id-section35).
* [Page 36:](#id-section36).
* [Page 37:](#id-section37).
* [Page 38:](#id-section38).
* [Page 39:](#id-section39).
* [Page 40:](#id-section40).
* [Page 41:](#id-section41).
* [Page 42:](#id-section42).
* [Page 43:](#id-section43).
* [Page 44:](#id-section44).
* [Page 45:](#id-section45).
* [Page 46:](#id-section46).
* [Page 47:](#id-section47).
* [Page 48:](#id-section48).
* [Page 49:](#id-section49).
* [Page 50:](#id-section50).
* [Page 51:](#id-section51).
* [Page 52:](#id-section52).
* [Page 53:](#id-section53).
* [Page 54:](#id-section54).
* [Page 55:](#id-section55).
* [Page 56:](#id-section56).
* [Page 57:](#id-section57).
* [Page 58:](#id-section58).
* [Page 59:](#id-section59).
* [Page 60:](#id-section60).

------
<div id='id-section1'/>
### Page 1: 2018-01-24. Notes on using Github, Rmarkdown, and the UNIX command-line

* Created github repos
* Began notebook
* Published notebook to github
* Log into UNIX server
* Reviewed basic BASH commands

------
<div id='id-section2'/>
### Page 2: 2018-01-29 Working with RNA-seq data

* Went over beetle experimental design
* covered outline of pipeline
* Use of the fastqc command to put fastqc data into home directory
fastqc WA_PP1_M2_GTCCGC_L003_R1_001.fastq.gz -o ~
*used winscp to transfer redulting file to computer


------
<div id='id-section3'/>
### Page 3:

Created clean reads directory in home
Transferred trim_example.sh to myscripts dir
Opened file with Vim 
* vim filename
vim commands:
*i = edit
*esc = exit edit
*: = enter commands
*w = write file
*q = quit vim
*:w newname = give file a new name
created trim_M2.sh
chmod u+x trim_M2.sh to make file executable
executed trim_M2.sh
fastqc on results
Mapping reads to reference transcriptome:
bwa mem <ref.fa> <read1.fq> <read2.fq> > <aln-pe.sam>
Used screen command
ctrl + a + d to detach

------
<div id='id-section4'/>
### Page 4:
Viewed .sam file with head and tailcommands
created tail file tail.sam
viewed tail file in vim
Reviewed sam flags
looked at python script to analyze sam file
Installed DESeq2 package in R
------
<div id='id-section5'/>
### Page 5:

Reviewed bash scripting making test_bash.sh (hello world!)
Moved aligned DESeq data to personal computer for work in R
Created thisthing.R 
------
<div id='id-section6'/>
### Page 6:

Continued thisthing.R with unusual results that differed from melissa
Visualized differences in gene expression across variables measures using PCA and ggplot
``` {r}
# set working directory
setwd("C:/Users/Paul/Desktop/Ecological_Genomics/DESeq_Data_and_scripts")

library("DESeq2")
library("ggplot2")

countsTable <- read.delim("allcountsdataRN_noIT.txt", header=TRUE, stringsAsFactors = TRUE, row.names=1)
countData <- as.matrix(countsTable)
head(countData)


conds <-  read.delim("cols_data_noIT.txt", header=TRUE, stringsAsFactors = TRUE, row.names=1)
head(conds)
colData <- as.data.frame(conds)

##################################

dds <- DESeqDataSetFromMatrix(countData = countData, colData, design = ~ devstage + sex + population) 

dim(dds)
#[1] 17483    48

dds <- dds[rowSums(counts(dds)) > 1,] #sum across columns but keep rows

dim(dds)
#[1] 16851    48  Only lost ~600 genes

dds <- DESeq(dds, modelMatrixType = "standard")

resultsNames(dds)

res <- results(dds)
str(res)

res <- res[order(res$padj),]
head(res)

# log2 fold change (MLE): population WA vs NC 
# Wald test p-value: population WA vs NC 
# DataFrame with 6 rows and 6 columns
# baseMean log2FoldChange     lfcSE      stat       pvalue         padj
# <numeric>      <numeric> <numeric> <numeric>    <numeric>    <numeric>
#   OTAU017482-RA 126.291608     -5.4166340 0.6732174 -8.045892 8.561964e-16 1.268369e-11
# OTAU012716-RA 188.877675      4.2644034 0.5427535  7.856980 3.935069e-15 2.914706e-11
# OTAU008667-RA 231.871115     -0.8736955 0.1267134 -6.895051 5.384538e-12 1.994164e-08
# OTAU012562-RA 251.774364     -0.8774079 0.1270957 -6.903520 5.072966e-12 1.994164e-08
# OTAU013988-RA   4.416955      4.4229857 0.6836393  6.469765 9.815559e-11 2.908154e-07
# OTAU011160-RA  10.241516     -2.5149390 0.4125305 -6.096371 1.085033e-09 2.678947e-06

summary(res)

# out of 16851 with nonzero total read count
# adjusted p-value < 0.1
# LFC > 0 (up)     : 333, 2% 
# LFC < 0 (down)   : 282, 1.7% 
# outliers [1]     : 85, 0.5% 
# low counts [2]   : 1952, 12% 
# (mean count < 1)
# [1] see 'cooksCutoff' argument of ?results
# [2] see 'independentFiltering' argument of ?results

res_pop <- results(dds, name="population_WA_vs_NC", alpha=0.05)

res_pop <- res_pop[order(res_pop$padj),]
summary(res_pop)

# out of 16851 with nonzero total read count
# adjusted p-value < 0.05
# LFC > 0 (up)     : 204, 1.2% 
# LFC < 0 (down)   : 177, 1.1% 
# outliers [1]     : 85, 0.5% 
# low counts [2]   : 1952, 12% 
# (mean count < 1)
# [1] see 'cooksCutoff' argument of ?results
# [2] see 'independentFiltering' argument of ?results

################################### Data Visualization
plotMA(res_pop, main="DESeq2", ylim=c(-2,2))
abline(h=c(-1,1), col="blue", lwd = 2)

# sex effect?

res_sex <- results(dds, name="sex_M_vs_F", alpha=0.05)
plotMA(res_sex, main="DESeq2", ylim=c(-15,15))
abline(h=c(-1,1), col="blue", lwd = 2)

##################### PCA to get a global view of gene expression
rld <- rlog(dds, blind=FALSE) #takes a while
data <- plotPCA(rld, intgroup=c("population","devstage","sex"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))

data$devstage <- factor(data$devstage, levels=c("L3L","PP1","PD1","AD4"), labels =c("L3L","PP1","PD1","AD4"))

ggplot(data, aes(PC1, PC2, color=sex, shape=devstage)) +
  geom_point(size=4, alpha=0.85) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  theme_minimal()
dev.off()

ggplot(data, aes(PC1, PC2, color=population, shape=devstage)) +
  geom_point(size=4, alpha=0.85) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  theme_minimal()

```
------
<div id='id-section7'/>
### Page 7:
Continued this thing.R in preparation for WGCNA analysis
``` {r}
################ the next several lines will make a cluster heat map of all the samples vs all samples

sampleDists <- dist(t(assay(rld)))

library("RColorBrewer")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(rld$population, rld$devstage, rld$sex, sep="-")
colnames(sampleDistMatrix) <- NULL

colors <- colorRampPalette(rev(brewer.pal(9,"Blues")))(255)

library("pheatmap")

pheatmap(sampleDistMatrix, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col=colors)

############### Let's look at individual genes!?!

d <- plotCounts(dds, gene="OTAU012716-RA", intgroup = (c("population", "sex", "devstage")), returnData = TRUE)
d

p <- ggplot(d, aes(x=devstage, y=count, shape=sex, color=population)) +
  theme_minimal() + theme(text = element_text(size=20), panel.grid.major = element_line(color="grey"))
p <- p + geom_point(position=position_jitter(w=0.3,h=0), size=3) +
  scale_x_discrete(limits=c("L3L","PP1","PD1","AD4"))
p

################### Save outputs

write.csv(res_pop, file="DGE_NCvsWA_pop.csv", row.names = T, quote=F)

### change pvalue to -log(pvalue) and export as .csv with rownames

write.csv(res, file="DGE_NCvsWA_pop.csv", row.names=T, quote=F) #this outputs results as csv file

neglogpval = as.matrix(-log(res$pvalue))   #script for GO Mann-Whitney test requires this format
head(neglogpval)
res_negpval = cbind(row.names(res), neglogpval)
colnames(res_negpval)=c("gene", "neglogpval")

write.csv(res_negpval, file="DGE_NCvsWA_pop_neglogpval.csv", row.names=F, quote=F)

############# Day 7 Transcriptomics 

colData$group <- factor(paste0(colData$population, "-", colData$devstage, "-", colData$sex))
head(colData)

dds <- DESeqDataSetFromMatrix(countData = countData, colData = colData, design = ~ group)
dds <- dds[rowSums(counts(dds)) > 1, ]
dim(dds)

dds <- DESeq(dds, parallel = T)

resultsNames(dds)

# [1] "Intercept"                  "group_NC.AD4.M_vs_NC.AD4.F"
# [3] "group_NC.L3L.F_vs_NC.AD4.F" "group_NC.L3L.M_vs_NC.AD4.F"
# [5] "group_NC.PD1.F_vs_NC.AD4.F" "group_NC.PD1.M_vs_NC.AD4.F"
# [7] "group_NC.PP1.F_vs_NC.AD4.F" "group_NC.PP1.M_vs_NC.AD4.F"
# [9] "group_WA.AD4.F_vs_NC.AD4.F" "group_WA.AD4.M_vs_NC.AD4.F"
# [11] "group_WA.L3L.F_vs_NC.AD4.F" "group_WA.L3L.M_vs_NC.AD4.F"
# [13] "group_WA.PD1.F_vs_NC.AD4.F" "group_WA.PD1.M_vs_NC.AD4.F"
# [15] "group_WA.PP1.F_vs_NC.AD4.F" "group_WA.PP1.M_vs_NC.AD4.F"

res_pop_PP1_F <- results(dds, contrast = list(c("group_WA.PD1.F_vs_NC.AD4.F"), c("group_WA.PD1.M_vs_NC.AD4.F")), listvalues = c(1/2, -1/2), alpha=0.05)

res_pop_PP1_F <- res_pop_PP1_F[order(res_pop_PP1_F$padj),]
head(res_pop_PP1_F)

summary(res_pop_PP1_F)

sig_pop_PP1_F <- res_pop_PP1_F[which(res_pop_PP1_F$padj <0.05), ]
dim(sig_pop_PP1_F) #check

sig_pop_PP1_F_df <- as.data.frame(sig_pop_PP1_F)
sig_pop_PP1_F_df$Row.names <- rownames(sig_pop_PP1_F_df)
dim(sig_pop_PP1_F_df)

genesOfInterest_pop_PP1_F <- c(sig_pop_PP1_F_df$Row.names)
length(genesOfInterest_pop_PP1_F)

vsd <- vst(dds, blind = FALSE)

#####

dds$combined = factor(paste0(dds$population, "-", dds$devstage, "-", dds$sex))
dds$combined <- factor(dds$combined, levels=c("WA-L3L-F","WA-L3L-M","WA-PP1-F","WA-PP1-M","WA-PD1-F","WA-PD1-M","WA-AD4-F","WA-AD4-M","NC-L3L-F","NC-L3L-M","NC-PP1-F","NC-PP1-M","NC-PD1-F","NC-PD1-M","NC-AD4-F","NC-AD4-M"), labels=c("WA-L3L-F","WA-L3L-M","WA-PP1-F","WA-PP1-M","WA-PD1-F","WA-PD1-M","WA-AD4-F","WA-AD4-M","NC-L3L-F","NC-L3L-M","NC-PP1-F","NC-PP1-M","NC-PD1-F","NC-PD1-M","NC-AD4-F","NC-AD4-M"))

baseMeanPerGrp <- sapply( levels(dds$combined), function(lvl) rowMeans( counts(dds,normalized=TRUE)[,dds$combined == lvl] ) )

head(baseMeanPerGrp)
dim(baseMeanPerGrp)

#pills out normalized counts (avg of 3 reps) for all of our sig genes
m <- baseMeanPerGrp[genesOfInterest_pop_PP1_F, c("WA-PP1-F", "WA-PP1-M", "NC-PP1-F", "NC-PP1-M")]

head(m)
dim(m)

mat_scaled = t(apply(m, 1, scale))
head(mat_scaled)

library(pheatmap)

pheatmap(mat_scaled, labels_col=c("WA-PP1-F", "WA-PP1-M", "NC-PP1-F", "NC-PP1-M"), cluster_cols = F, cluster_rows = T)

########### 

# Export counts to use as the input for WGCNA

norm.counts <- counts(dds, normalized=TRUE)
dim(norm.counts)

write.csv(norm.counts, file = "beetle_norm_counts.csv", row.names=T, quote=F)


````
 Created an R script for WGCNA (WGCNA.R):
``` {r}
#WGCNA script

source("http://bioconductor.org/biocLite.R") 
biocLite(c("AnnotationDbi", "impute", "GO.db", "preprocessCore")) 
install.packages("WGCNA")

library("WGCNA")

setwd("C:/Users/Paul/Desktop/Ecological_Genomics/DESeq_Data_and_scripts")

beetData = read.csv("beetle_norm_counts.csv")
dim(beetData)
head(beetData)
names(beetData)

bdatExpr0 = as.data.frame(t(beetData[, -c(1:1)]))
dim(bdatExpr0)
names(bdatExpr0) = beetData$X

rownames(bdatExpr0) = names(beetData)[-c(1:1)]

bgsg = goodSamplesGenes(bdatExpr0, verbose = 3)

############ Make acluster tree to look for outlier samples

sampleTree =hclust(dist(bdatExpr0), method = "average")
##plot
sizeGrWindow(12,9)
par(cex = 0.6)
par(mar=c(0,4,2,0))

plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab =1.5, cex.axis = 1.5, cex.main =2)

########### Read in trait data

btraitData = read.table("cols_dat_noIT_num[1920].txt", header = TRUE)


##### Network construction and module detection

powers = c(c(1:10), seq(from =12, to=20, by=2))
#call the network topology function

sft = pickSoftThreshold(bdatExpr0, powerVector = powers, verbose=5)

#plot results
sizeGrWindow(9,5)
par(mfrow = c(1,2))
cex1 = 0.9

#scale-free topology fit index as a function of the soft-thresholding power

plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndeices[,2], xlab="softThreshold (power)", ylab= "Scale Free Topology Model Fit, signed R^2", type="n",main=paste("Scale independence"))

text(sft$fitIndices[1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2], labels=powers, cex=1, col="red")

abline(h=0.9,col="red")

# mean connectivity as a function of the soft-thresholding power

plot(sft$fitIndices[,1], sft$fitIndices[,5], xlab="Soft Threshold (power)", ylab="Mean Connectivity", type="n", main=paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1, col="red")

#################

net = blockwiseModules(bdatExpr0, power=6, TOMType = "unsigned", miniModuleSize =30, reassignThreshold = 0, mergeCutHeight = 0.25, numericLabels = TRUE, pamRespectsDendro = FALSE, saveTOMs = T, saveTOMFileBase = "beetleTOM", verbose=3)

sizeGrWindow(12,9)
mergedColors = labels2colors(net$colors)

## Plot the dendrogram with the module colors underneath
plotDendroAndColors(net$dendrograms[[1]], mergedColors[net$blockGenes[[1]]], "Module Colors", dendroLables = F, hang = 0.03, addGuide = T, guideHang = 0.05)

moduleLabels = net$colors
moduleColors = labels2colors(net$colors)
MEs = net$MEs
geneTree = net$dendrograms[[1]]

####

nGenes = ncol(bdatExpr0)
nSamples = nrow(bdatExpr0)

MEs0 = moduleEigengenes(bdatExpr0, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = cor(MEs, btraitData, use = "p")

moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples)

sizeGrWindow(10,6)
textMatrix = paste(signif(moduleTraitCor, 2), "\n(", signif(moduleTraitPvalue, 1), ")", sep = "")
dim(testMatrix) = dim(moduleTraitCor)
par(mar = c(6, 8.5,3,3))
labledHeatmap(MAtrix=moduleTraitCor, xlabels= names(btraitData), ylabels = names(MEs), ySymbols = names(MEs), colorlabels = F, colors = greenWhiteRed(50), textMatrix = textMatrix, setStdMargins = F, cex.text = o.5, zlin = c(-1,1), main = paste("Module-trait relationships"))

```
------
<div id='id-section8'/>
### Page 8: 
See above. Consolidated everything into one script.
------
<div id='id-section9'/>
### Page 9:

Beginning of Genomics Section

Methodology:

Return to sam fies (cleaned reads aligned to reference genome)

Call SNPs and Genotypes using read2snps and samtools as pipelines

Quality Control using vcftools

Calculate diversity statistics (Tajima's D, Structure, Selection, etc.)

Created CallSNPs.sh

Used samtools tview [file] [path to reference genome] to view reads on genome


------ 
<div id='id-section10'/>
### Page 10:
Diversity and site frequency spectrum
used vcftools to augment vcf file
 vcftools --vcf OTAU_2018_samtools.vcf --min-alleles 2 --max-alleles 2 --maf 0.01 --minGQ 10 --max-missing 0.8 --het --out ~/mydata/samtools0.8v
------
<div id='id-section11'/>
### Page 11:
Missed Day
Found Jamie's notes very useful to catch up:
https://github.com/JKostyun/2018_Ecological_Genomics_Notebook_JLK/blob/master/Template_online_notebook.Rmd#id-section12
------
<div id='id-section12'/>
### Page 12:
Missed Day
Again, Jamie's notes were quite good for catching up:
https://github.com/JKostyun/2018_Ecological_Genomics_Notebook_JLK/blob/master/Template_online_notebook.Rmd#id-section13
------
<div id='id-section13'/>
### Page 13:
Day 5 of genomics
Continuing on admixture analysis
Copied data and bash scripts to myspcripts folder
Found k values for populations
Created R script ADMIXTURE.R to visulaize K with k= 2,3,4
Homework: CHoose a different thinning setting for the snps (hgher or lower)
          create a new vcf file
          rerun admixture with steps followed today
          assess "robustness" of our inference of K from today (how much does our view depend on how we account for linkage disequilibrium)
          
          For each population (IT, NC, WA)
            Calculate pi(nucleotide diversity) IT>NC,WA?
              vcftools --vcf filename.vcf --keep popfile --site-pi
              
          Propose a hypothesis for the demographic history
            
------
<div id='id-section14'/>
### Page 14:
Population Genomics 6: Testing selection using Fst outlier analysis
Using bayscan to find local adaptation
bayscan parameters: pr_odds 10 = one in every ten snps under local selection
can have huge effect on results. We will very this parameter within the class and compare results. I have 500. Also verying thin parameter between 10 20 and 30.
Wrote a bayescan.sh ship that did this in mycripts
Screen to put in screen
ctrl+a+d to detach
screen -r to reattach

------
<div id='id-section15'/>
### Page 15:
Population Genomics 7: Analyzing Fst outlier (Bayescan) results
Investigating our Fst file
prob high = selection low = neutral
log10.Po - similar
qval like p value
WOrked in R to look at Fst outliers
Exported candidate genes to the server
Looked at annotation file OTAU.Models.gff3
------
<div id='id-section16'/>
### Page 16:
```{r}
setwd("C:/Users/Paul/Desktop/Ecological_Genomics/Genomic_Homework/")

library(pophelper) #Pophelper makes nice looking figures with MANY customizable settings.

admixfiles=list.files(path=("ADMIXTURE/"),full.names=T)
admixlist=readQ(files=admixfiles,filetype="basic")

# metadata: sample id and pop from beetle.pop file
metadata=read.table("cols_data.txt",header=T)

# format metadata to a data frame and ind variables as chars. for plotting
metadata2=data.frame(sampleid=metadata[,1], population=metadata[,2])

metadata2$sampleid=as.character(metadata2$sampleid)
metadata2$population=as.character(metadata2$population)

# add in the sample id to the different Q files for plotting
if(length(unique(sapply(admixlist,nrow)))==1)
  admixlist <- lapply(admixlist,"rownames<-",metadata2$sampleid)

head(admixlist[[3]])

#Make ancestry plots grouped by population

p <- plotQ(admixlist[c(1,2,3)],
           returnplot=T,exportplot=T,quiet=T,basesize=11, imgoutput="join", 
           showyaxis=T, showticks=T, panelspacer=0.4, useindlab=F, showindlab=F, 
           grplab=metadata2[2], grplabsize=4, linesize=1, barsize=1, pointsize=3, 
           panelratio=c(4,1), divsize = 0.75, pointcol="white", showtitle=T, 
           titlelab="ADMIXTURE analysis on O. tauri, SNPs thinned to 0.1 per kb", 
           splab=c("K=2","K=3","K=4"), outputfilename="ADMIXTURE_Otauri_100_thin",
           imgtype="pdf",height=3,width=25)

plot(p$plot[[1]])

IT_pi = read.table("IT_pi.sites.pi", header=T)



```
------
<div id='id-section17'/>
### Page 17:
All scripts on server (~/data/projectdata/beetles/share/Demhist)
------
<div id='id-section18'/>
### Page 18:
------
<div id='id-section19'/>
### Page 19:
------
<div id='id-section20'/>
### Page 20:
------
<div id='id-section21'/>
### Page 21:
------
<div id='id-section22'/>
### Page 22:
------
<div id='id-section23'/>
### Page 23:
------
<div id='id-section24'/>
### Page 24:
------
<div id='id-section25'/>
### Page 25:
------
<div id='id-section26'/>
### Page 26:
------
<div id='id-section27'/>
### Page 27:
------
<div id='id-section28'/>
### Page 28:
------
<div id='id-section29'/>
### Page 29:
------
<div id='id-section30'/>
### Page 30:
------
<div id='id-section31'/>
### Page 31:
------
<div id='id-section32'/>
### Page 32:
------
<div id='id-section33'/>
### Page 33:
------
<div id='id-section34'/>
### Page 34:
------
<div id='id-section35'/>
### Page 35:
------
<div id='id-section36'/>
### Page 36:
------
<div id='id-section37'/>
### Page 37:
------
<div id='id-section38'/>
### Page 38:
------
<div id='id-section39'/>
### Page 39:
------
<div id='id-section40'/>
### Page 40:
------
<div id='id-section41'/>
### Page 41:
------
<div id='id-section42'/>
### Page 42:
------
<div id='id-section43'/>
### Page 43:
------
<div id='id-section44'/>
### Page 44:
------
<div id='id-section45'/>
### Page 45:
------
<div id='id-section46'/>
### Page 46:
------
<div id='id-section47'/>
### Page 47:
------
<div id='id-section48'/>
### Page 48:
------
<div id='id-section49'/>
### Page 49:
------
<div id='id-section50'/>
### Page 50:
------
<div id='id-section51'/>
### Page 51:
------
<div id='id-section52'/>
### Page 52:
------
<div id='id-section53'/>
### Page 53:
------
<div id='id-section54'/>
### Page 54:
------
<div id='id-section55'/>
### Page 55:
------
<div id='id-section56'/>
### Page 56:
------
<div id='id-section57'/>
### Page 57:
------
<div id='id-section58'/>
### Page 58:
------
<div id='id-section59'/>
### Page 59:
------
<div id='id-section60'/>
### Page 60:

------
